git-branchless/src/commands/reword.rs

```diff
@@ -90,6 +93,43 @@ pub fn reword(
         }
     };
 
+    let messages = match messages {
+        InitialCommitMessages::Discard | InitialCommitMessages::Messages(_) => messages,
+        InitialCommitMessages::FixUp(revset) => {
+            let commits_to_fixup =
+                resolve_commits_from_hashes(&repo, &mut dag, effects, vec![revset.clone()])?
+                    .unwrap_or_default();
+            let commit_to_fixup = match commits_to_fixup.as_slice() {
+                [commit_to_fixup] => {
+                    if commits.iter().any(|commit| {
+                        !dag.query()
+                            .is_ancestor(commit_to_fixup.get_oid().into(), commit.get_oid().into())
+                            .expect("Failed checking ancestry of commit.")
+                    }) {
```

Comment:
I suppose that instead of `iter().any(...)` I could instead just use a single dag query; perhaps like
```rust
commits.intersection(
    dag.query.descendents(
        commit_to_fixup.get_oid().into()
    )
) == commits
```

Reply:
Nevermind this; I thought that `NameSet` allowed for `==/!=` but it doesn't. I'll leave it as is unless you have a suggestion for a more intuitive way to check this.

Reply:
I think you could write this as `dag.query().common_ancestors(CommitSet::from(commits))?.contains(commit_to_fixup)?` or something like that. I don't know why there isn't a `.equals()` method or similar.